### ISSUE TITLE
Components.Server is incorrectly built on linux-musl build

### ISSUE THREAD

**Issue by pranavkm** [aspnet/AspNetCore#8861]
Components.Server.csproj [conditionally embeds](https://github.com/aspnet/AspNetCore/blob/master/src/Components/Server/src/Microsoft.AspNetCore.Components.Server.csproj#L36) the components.server.js based on the presence of a flag. However, this file is absolutely required for the server to work.

As of now, `BuildNodeJS=false` is true in the linux-musl build. Need to figure out why it's set for that build configuration and what we need to do to fix this.

Note: This is not a customer facing issue as long as we ship Components.Server as a package. Once it's part of the shared framework, we absolutely need to address this.

**Comment by pranavkm** [comment]
cc @SteveSandersonMS

**Comment by SteveSandersonMS** [comment]
Thanks for tracking this, @pranavkm!

**Comment by pranavkm** [comment]
@JunTaoLuo \ @dougbu, would you know the answer to this?

**Comment by dougbu** [comment]
I'm not seeing where `-BuildNodeJS` or `--build-nodejs` is passed into the build script and the default is not to build NodeJS projects. Where we use `-All` or `--all`, we also specify `-NoBuildNodeJS` or `--no-build-node-js`.

So, I'm not sure where we're building NodeJS projects. Is this what you're seeing?

**Comment by pranavkm** [comment]
Components.Server depends on https://github.com/aspnet/AspNetCore/blob/master/src/Components/Browser.JS/Microsoft.AspNetCore.Components.Browser.JS.npmproj to have successfully built. You're right when you call out that `-NoBuildNodeJs` is set when the build says `-All`, I thought that was limited to the linux-musl configuration.

I modified the Components.Server project to always build npmproj and noticed that it was failing on the linux-musl configuraiton:  https://github.com/aspnet/AspNetCore/runs/85530564 when we invoked `yarn install`. Any guesses why that might be the case?
